src/api/local-storage-provider.model.ts

```diff
@@ -0,0 +1,76 @@
+import { ScheduleDataModel } from "../common-models/schedule-data.model";
+import { ShiftInfoModel, ShiftModel } from "../common-models/shift-info.model";
+import { WorkerInfoModel, WorkersInfoModel } from "../common-models/worker-info.model";
+import { ScheduleDataActionType } from "../state/reducers/schedule-data.reducer";
+import {
+  PersistanceStoreProvider,
+  RevisionDescription,
+  ScheduleRevision,
+  ThunkFunction,
+  TimeUnit,
+} from "./persistance-store.model";
+
+export class LocalStorageProvider implements PersistanceStoreProvider {
+  private sleep(ms: number) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+  saveScheduleRevision(revision: ScheduleRevision): ThunkFunction<ScheduleDataModel> {
+    return async (dispatch, getState) => {
```

Comment:
W skrócie:
redux pozwala dodawać do stanu różne middleware. Jednym z takich jest `redux-thunk`. Ono robi to, że zanim zdispatchuje jakąś akcję, sprawdza typ obiektu który był przekazany jako akcja i w przypadku jeżeli jest funkcją - wywołuje tą funckję podając jako parametry `dispatch` i `getState`.
```
// Write a synchronous outer function that receives the `text` parameter:
export function saveNewTodo(text) {
  // And then creates and returns the async thunk function:
  return async function saveNewTodoThunk(dispatch, getState) {
    // ✅ Now we can use the text value and send it to the server
    const initialTodo = { text }
    const response = await client.post('/fakeApi/todos', { todo: initialTodo })
    dispatch({ type: 'todos/todoAdded', payload: response.todo })
  }
}
```

Tu jest podany bardziej szczególowy opis:
https://redux.js.org/tutorials/fundamentals/part-6-async-logic#saving-todo-items